photon-lib/src/main/scala/com/linkedin/photon/ml/hyperparameter/search/GaussianProcessSearch.scala

```diff
@@ -56,10 +57,11 @@ class GaussianProcessSearch[T](
     evaluationFunction: EvaluationFunction[T],
     evaluator: Evaluator,
     discreteParams: Seq[Int] = Seq(),
+    indexedTransformMap: Map[Int, String] = Map(),
```

Comment:
Let's pass the kernel here instead of the transform map. `GaussianProcessSearch` really operates at the "kernel" level of specificity, not the "value" level. It should default to `new Matern52`.